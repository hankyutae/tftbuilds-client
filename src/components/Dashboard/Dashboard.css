.Dashboard{
  padding:18px 20px;
  text-align:left;
  padding-top:30px;
}
.display-account-builds-box{
  background-color:rgb(155, 200, 87);
  padding-left:15px;
  padding-right:15px;
  padding-bottom:10px;
  padding-top:1px;
  margin-top:20px;
}