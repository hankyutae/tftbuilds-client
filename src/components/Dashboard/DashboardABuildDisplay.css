.build-display-in-dashboard{
  position:relative;
  display:flex;
  flex-wrap:wrap;
  align-items:top;
  background-color:whitesmoke;
}
.build-display-in-dashboard-box{
  position:relative;
  background-color:whitesmoke;
  margin:10px 0px;
}
.load-and-share{
  display:flex;
  flex-direction: column;
  justify-content: start;
  align-items:left;
}
.dash-delete-build{
  position:absolute;
  top:0;
  right:0;
}
.build-share-link{ 
  overflow-wrap: break-word;
  word-wrap: break-word;

  -ms-word-break: break-all;
  /* This is the dangerous one in WebKit, as it breaks things wherever */
  word-break: break-all;
  /* Instead use this non-standard one: */
  word-break: break-word;
}