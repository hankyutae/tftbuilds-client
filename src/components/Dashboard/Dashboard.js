import React from 'react';
import './Dashboard.css';
import TftBuildsApiService from '../../services/tftbuilds-api-service';
import DashboardABuildDisplay from './DashboardABuildDisplay';

export default class Dashboard extends React.Component {
  state = {
    allbuilds: [],
    removeABuild: (index) => {
      let oneLvlDeepCopy = [...this.state.allbuilds];
      oneLvlDeepCopy.splice(index, 1);
      this.setState({
        allbuilds: oneLvlDeepCopy
      })
    }
  }
  componentDidMount() {
    TftBuildsApiService.getBuilds()
      .then(builds => {
        this.setState({
          allbuilds: builds
        })
      });
  }
  render() {
    return (
      <div className='Dashboard h2-nav-is-above'>
        {/* <h2>Your Account</h2>
        <div className='account-stuff'>

        </div> */
          <div className='display-account-builds-box'>
            
            <h2>Your Builds</h2>
        
            {this.state.allbuilds.length !== 0 &&
              this.state.allbuilds.map((build, index) => {
                return <DashboardABuildDisplay key={index} build={build} index={index} handleRemove={this.state.removeABuild} />
              })
            }
          </div>
        }

      </div>
    )
  }
}