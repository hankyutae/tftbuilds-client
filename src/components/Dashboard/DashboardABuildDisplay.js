import React from 'react';
import './DashboardABuildDisplay.css'
import ImgLink from '../../services/create-img-link-service';
import { Link } from 'react-router-dom';
import TftContext from '../../contexts/TftContext';
import TftBuildsApiService from '../../services/tftbuilds-api-service';

export default class DashboardABuildDisplay extends React.Component {
  state = {
    isExpanded: false,
    isRevealed: false,
    copySuccess: '',
    isDeleting:false,
  }
  static contextType = TftContext
  handleLoadBuildClick = (e) => {
    e.stopPropagation();
    this.context.setCurrentBuild(this.props.build['build_data']);
  }
  componentDidUpdate(prevProps){
    /* this.setState({
      isExpanded: false,
      isRevealed: false,
      copySuccess: '',
      isDeleting:false,
    }) */
  }
  handleBuildClick = (e) => {
    e.stopPropagation();
    this.setState({
      isExpanded: !this.state.isExpanded
    })
  }
  handleDeleteClick = (e) => {
    e.stopPropagation();
    TftBuildsApiService.deleteBuild(this.props.build.id)
      .then(res => {
        this.setState({
          isExpanded: false,
          isRevealed: false,
          copySuccess: '',
          isDeleting:false,
        },()=>{
          this.props.handleRemove(this.props.index);
        })
      })
  }
  toggleDeleting = (e)=>{
    this.setState({
      isDeleting:!this.state.isDeleting
    })
  }
  handleMakePublicClick = (e) => {
    e.stopPropagation();
    if (this.props.build['is_public']) {
      this.setState({
        isRevealed: true
      })
    }
    else {
      TftBuildsApiService.makeBuildPublic(this.props.build.id)
        .then(res => {
          this.setState({
            isRevealed: true
          },this.props.handleMakePublic(this.props.index))
        })
    }
  }
  copyLink =(e)=>{
    this.textArea.select();
    document.execCommand('copy');
    this.setState({copySuccess:'Copied!'})
  }
  render() {
    return (
      <div className='build-display-in-dashboard-box'>
        {Object.keys(this.context.champions).length !== 0 && Object.keys(this.context.items).length !== 0 &&
          <>
            <div className='dashboard-build-top'>
              <div className='load-and-share'>
                <div className='load-build-button button-hover-lighten'>
                  <Link className='nav-link' to='/create-build' onClick={this.handleLoadBuildClick}> Load Build</Link>
                </div>
                <div className='share-build-button button-hover-lighten' onClick={this.handleMakePublicClick}>
                  Share
                </div>
              </div>
              <div className='dash-delete-build'>
                {
                  this.state.isDeleting? 
                  <div className='delete-yes-or-no-container'>
                    <div className='yes-delete-build-button deleting-yes-or-no-button button-hover-lighten'onClick={this.handleDeleteClick}>
                      Delete
                    </div>
                    <div className='no-delete-build-button deleting-yes-or-no-button button-hover-lighten' onClick={this.toggleDeleting}>
                      No
                    </div>
                  </div>
                  :
                  <div className='red delete-build-button button-hover-lighten' onClick={this.toggleDeleting}>Delete </div>
                }
                
              </div>

            </div>
            <div className='dashboard-build-bottom'>

            
              <div className='build-display-in-dashboard' onClick={this.handleBuildClick}>
                {




                  this.props.build['build_data'].filter((champ, index) => index < 11).map((champ, index) => {
                    return (
                      <div key={index} className='dash-build-champ-details'>
                        <img key={index} className='dash-build-champ-icon' src={ImgLink.createLink(this.context.champions[champ.id].splash)} alt={this.context.champions[champ.id].name + ' icon'} />
                        <div className='dash-box-champ-items'>
                          {champ.items.map((item,index)=>{
                            return <img key={index} className='dash-build-item-icon' src={ImgLink.createLink(this.context.items[item].icon)} alt={this.context.items[item].name + ' icon'} />
                          })}
                        </div>
                      </div>
                    )
                  })

                }
              </div>

              {this.state.isRevealed &&
                <div className='build-link-container'>
                  <textarea 
                    className='build-share-link' 
                    ref={(textarea) => this.textArea = textarea} 
                    value={`${window.location.origin}/build/${this.props.build.id}`}
                    readOnly
                  />
                  {
                    document.queryCommandSupported('copy') &&
                    <div className='copy-link-button' onClick={this.copyLink}>
                      Copy
                    </div>
                  }
                  {
                    this.state.copySuccess &&
                    <span className='copy-success'>
                      {this.state.copySuccess}
                    </span>
                  }

                </div>
              }
            </div>
          </>
        }
      </div>
    )
  }
}