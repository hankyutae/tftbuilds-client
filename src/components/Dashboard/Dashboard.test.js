import React from 'react';
import ReactDOM from 'react-dom';
import Dashboard from './Dashboard';
import renderer from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
      <Dashboard/>
  , div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create( 
      <Dashboard/>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});