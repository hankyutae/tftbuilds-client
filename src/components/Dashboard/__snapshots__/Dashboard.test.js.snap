// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Dashboard h2-nav-is-above"
>
  <div
    className="display-account-builds-box"
  >
    <h2>
      Your Builds
    </h2>
  </div>
</div>
`;
